Source/Scene/MetadataClassProperty.js

```diff
@@ -679,6 +737,15 @@ MetadataClassProperty.prototype.packVectorAndMatrixTypes = function (
  * @private
  */
 MetadataClassProperty.prototype.validate = function (value) {
+  if (!defined(value) && defined(this._default)) {
+    // no value, but we have a default to use.
+    return undefined;
+  }
+
+  if (this._required && !defined(value)) {
+    return `required property must have a value`;
+  }
```

Comment:
Should these two statements be swapped? If the property is required it should error out even if it has a default value.

This should also be enforced at the spec level. `default` and `noData` should only be allowed if `required` is false. I'll add that clarification.